File: haskell/src/Solutions/Euler235.hs
module Solutions.Euler235 where
import Utils.Numeric(bisection)
import Text.Printf

f :: Double -> Double
f r = let s = r^5000 in 300 * (s - 1) * (r - 1) - s * (r * 5000 - 5001) - 1 + (r - 1)^2 * 200000000000

euler235 :: IO String
euler235 = return $ printf "%.12f" $ bisection f 40 1.001 1.003

File: rust/src/solutions/euler235.rs
fn f(r: f64) -> f64 {
    let s = r.powi(5000);
    300.0 * (s - 1.0) * (r - 1.0) - s * (r * 5000.0 - 5001.0) - 1.0
        + (r - 1.0).powi(2) * 200000000000.0
}

pub fn euler235() -> String {
    format!(
        "{:.12}",
        crate::utils::numeric::bisection::bisection(f, 1.001, 1.003, 40)
    )
}
